src/AddIns/Uno.UI.Lottie.Wasm/ts/Uno.UI.Lottie.ts

```diff
@@ -0,0 +1,255 @@
+
+				Lottie.updateProperties(currentAnimation, newProperties);
+			});
+
+			return "ok";
+		}
+
+		public static stop(elementId: number): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.stop();
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static play(elementId: number, fromProgress: number, toProgress: number, looped: boolean): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.loop = looped;
+
+				const fromFrame = fromProgress * Lottie._numberOfFrames;
+				const toFrame = toProgress * Lottie._numberOfFrames;
+
+				a.playSegments([fromFrame, toFrame], false);
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static kill(elementId: number): string {
+			Lottie.withPlayer(p => {
+				Lottie._runningAnimations[elementId].animation.destroy();
+				delete Lottie._runningAnimations[elementId];
+			});
+
+			return "ok";
+		}
+
+		public static pause(elementId: number): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.pause();
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static resume(elementId: number): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.play();
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static setProgress(elementId: number, progress: number): string {
+			Lottie.withPlayer(p => {
+				const animation = Lottie._runningAnimations[elementId].animation;
+				const frames = animation.getDuration(true);
+				const frame = frames * progress;
+				animation.goToAndStop(frame, true);
+				Lottie.raiseState(animation);
+
+			});
+
+			return "ok";
+		}
+
+		public static getAnimationState(elementId: number): string {
+			const animation = Lottie._runningAnimations[elementId].animation;
+
+			const state = Lottie.getStateString(animation);
+
+			return state;
+		}
+
+		private static needNewPlayerAnimation(current: LottieAnimationProperties, newProperties: LottieAnimationProperties): boolean {
+
+			if (current.jsonPath !== newProperties.jsonPath) {
+				return true;
+			}
+			if (newProperties.stretch !== current.stretch) {
+				return true;
+			}
+			if (newProperties.autoplay !== current.autoplay) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private static updateProperties(
+			runningAnimation: RunningLottieAnimation,
+			newProperties: LottieAnimationProperties) {
+
+			const animation = runningAnimation.animation;
+			const runningProperties = runningAnimation.properties;
+
+			if (runningProperties == null || newProperties.rate != runningProperties.rate) {
+				animation.setSpeed(newProperties.rate);
+			}
+
+			runningAnimation.properties = newProperties;
+		}
+
+		private static createAnimation(properties: LottieAnimationProperties, animationData?: AnimationData): RunningLottieAnimation {
```

Comment:
Codacy found an issue: [Exceeds maximum line length of 120](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)